Improve dependency checking

ehaabbasil opened the pull request:
#1 

Added `i show_deps` to analyze intra-directory dependencies in `helpers`, with text/DOT reports, tests (94% coverage), and updated docs.

- Implemented `i show_deps` in `tasks.py` with `max_level` and `show_cycles`

- Added `import_check/dependency_graph.py` for dependency analysis
- Added tests in `import_check/test/test_dependency_graph.py` 

- Created script `generate_deps.py` as a fallback for `invoke` failures

- Updated `docs/work_tools/all.import_check.reference.md` with usage
- Updated `.gitignore` to exclude generated files

- Used `nx.nx_pydot.write_dot`

- Isolated tests due to `ModuleNotFoundError` for `helpers.hdbg` and `helpers.hunit_test` in `conftest.py,` likely due to missing dependencies; used a temporary directory to avoid loading `conftest.py`

- Fixed indentation error in `helpers/lib_tasks_docker.py`

- Fixed `CSFY_ECR_BASE_PATH` issue using `os.environ.get()`

- Created PR in fork due to permission restrictions in causify-ai/helpers

- FYI: @gpsaggese @sonniki @Shaunak01

ehaabbasil committed:
Add .gitignore for generated files

heanhsok committed:
HelpersTask634_Centralize_install_module_if_not_present_in_one_of_the_helpers (#642)

* Checkpoint

* checkpoint

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

heanhsok committed:
CmampTask12028_Build_fail___cmamp___run_slow_tests_14744852229 (#643)

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' skipped
All checks passed ✅

* Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

ehaabbasil committed:
Enhance dependency checking with i show_deps

A new DependencyGraph class for building, analyzing and visualizing dependency graphs.
Added Unit Tests, achieved 94% coverage for dependency_graph.py.
Updated documentation in docs/work_tools/all.import_check.reference.md
A fallback script (generate_deps.py) for generating dependency graphs when invoke fails. (work around pipeline)

Shaunak01 committed:
HelpersTask597_Configure_Codecov_for_coverage (#598)

* testing

* testing

* testing

* testing

* testing

* testing

* testing

* testing

* testing

* testing

* checkpoint

* checkpoint

* checkpoint

* test with superslow

* test

* test

* test

* test

* flag test

* flag test

* flag test

* test

* test

* test

* test

* checkpoint

* test

* test

* test

* checkpoint

* test

* test

* test

* test

* test

* checkpoint

* checkpoint

* test

* test

* checkpoint

* test

* test

* test

* test

* test

* test

* debug

* test

* test

* test

* checkpoint

* checkpoint

* Test

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Test

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Test

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

---------

Co-authored-by: Vlad <dremdem@gmail.com>

sandeepthalapanane committed:
HelpersTask615_Come_up_with_a_testing_approach_for_release_flow (#640)

* HelpersTask615: Add unittest for docker_build_local_image()

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Modify comments and docstring

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Move helper functions out of the class

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Rename helper functions

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Split the check into seperate checks

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Remove .body from function call

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Update

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Propogate changes from previous commit

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* HelpersTask615: Modify comments

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

---------

Co-authored-by: Vlad <dremdem@gmail.com>
Co-authored-by: GP Saggese <saggese@gmail.com>
Co-authored-by: GP Saggese <33238329+gpsaggese@users.noreply.github.com>

heanhsok committed:
HelpersTask634_Centralize_install_module_if_not_present_in_one_of_the_helpers_2 (#650)

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' skipped
All checks passed ✅

* Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

heanhsok committed:
HelpersTask622_gitleaks_doesnt_run_in_submodules (#654)

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Remove temp file

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

heanhsok committed:
HelpersTask644_Bug_in_go_XYZ.py (#653)

* Checkpoint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Improve

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' skipped
All checks passed ✅

* Add comment

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' skipped
All checks passed ✅

* Update

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' skipped
All checks passed ✅

* Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

gpsaggese committed:
Lint

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_words' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

gpsaggese commented:
Can you pls add me to your fork so I can push some changes

```
> git push
...
To github.com:ehaabbasil/helpers.git
 ! [remote rejected] improve-dependency-checking -> improve-dependency-checking (permission denied)
error: failed to push some refs to 'github.com:ehaabbasil/helpers.git'
```

ehaabbasil commented:
@gpsaggese Done

gpsaggese commented:
Still no luck, in fact
```
> gh repo view ehaabbasil/helpers --json viewerPermission
{
  "viewerPermission": "READ"
}
```

gpsaggese commented:
Now it's ok. GH needed some time to propagate the perms

```
> gh repo view ehaabbasil/helpers --json viewerPermission
{
  "viewerPermission": "WRITE"
}
```

gpsaggese committed:
CmampTask11862_Allow_dind_unit_tests_to_run_on_server_and_CI_4 (#658)

* Update

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

* Update

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅

gpsaggese committed:
Merge remote-tracking branch 'origin/master' into improve-dependency-checking

No pre-commit output found.

gpsaggese committed:
Update

Pre-commit checks:
- 'check_master' passed
- 'check_author' passed
- 'check_file_size' passed
- 'check_python_compile' passed
- 'check_gitleaks' passed
All checks passed ✅
